app/src/main/java/com/infomaniak/mail/data/cache/mailboxInfos/MailboxController.kt

```diff
@@ -45,32 +52,34 @@ object MailboxController {
     }
 
     private fun MutableRealm?.getMailboxesQuery(userId: Int): RealmQuery<Mailbox> {
-        return (this ?: RealmDatabase.mailboxInfos).query("${Mailbox::userId.name} == '$userId'")
+        return (this ?: RealmDatabase.mailboxInfos).query("${Mailbox::userId.name} = '$userId'")
     }
 
     fun getMailbox(objectId: String, realm: MutableRealm? = null): Mailbox? {
         return realm.getMailboxQuery(objectId).find()
     }
 
-    private fun getMailboxAsync(objectId: String, realm: MutableRealm? = null): SharedFlow<SingleQueryChange<Mailbox>> {
-        return realm.getMailboxQuery(objectId).asFlow().toSharedFlow()
+    private fun MutableRealm?.getMailboxQuery(objectId: String): RealmSingleQuery<Mailbox> {
+        return (this ?: RealmDatabase.mailboxInfos).query<Mailbox>("${Mailbox::objectId.name} = '$objectId'").first()
     }
 
-    private fun MutableRealm?.getMailboxQuery(objectId: String): RealmSingleQuery<Mailbox> {
-        return (this ?: RealmDatabase.mailboxInfos).query<Mailbox>("${Mailbox::objectId.name} == '$objectId'").first()
+    fun getCurrentMailbox(): Mailbox? = with(RealmDatabase.mailboxInfos) {
+        val checkUserId = "${Mailbox::userId.name} = '${AccountUtils.currentUserId}'"
+        val checkMailboxId = "${Mailbox::mailboxId.name} = '${AccountUtils.currentMailboxId}'"
+        val mailbox = query<Mailbox>("$checkUserId AND $checkMailboxId").first().find()
+        return mailbox ?: query<Mailbox>(checkUserId).first().find()
     }
     //endregion
 
     //region Edit data
-    fun update(apiMailboxes: List<Mailbox>): List<Mailbox> {
+    fun update(apiMailboxes: List<Mailbox>) {
 
         // Get current data
         Log.d(RealmDatabase.TAG, "Mailboxes: Get current data")
         val realmMailboxes = getMailboxes(AccountUtils.currentUserId)
 
         // Get outdated data
         Log.d(RealmDatabase.TAG, "Mailboxes: Get outdated data")
-        // val deletableMailboxes = MailboxInfoController.getDeletableMailboxes(apiMailboxes)
         val deletableMailboxes = realmMailboxes.filter { realmMailbox ->
             apiMailboxes.none { apiMailbox -> apiMailbox.mailboxId == realmMailbox.mailboxId }
         }
```

Comment:
No manual operation if we can use a realm query, and is it really necessary ?

Reply:
Can't succeed to do it for now using `IN` realm query 😢

Reply:
To see together then

Reply:
Done.